src/random_test.cpp

```diff
@@ -1215,7 +1215,8 @@ bool execute_sql(std::string sql, Thd1 *thd) {
     if (log_all || log_success) {
       thd->thread_log << " S " << sql;
       if (result == NULL) {
```

Comment:
int affected_row = result == NULL ? mysql_affected_rows(thd->conn) : mysql_num_rows(result);
thd->thread_log << " rows:" << affected_row << std::endl;

Reply:
It cannot be done this way since result is defined as:
MYSQL_RES *result;
I have done it this way:
    /* log successful query */
    if (log_all || log_success) {
      thd->thread_log << " S " << sql;
      int number;
      if (result == NULL)
        number = mysql_affected_rows(thd->conn);
      else
        number = mysql_num_rows(result);
      thd->thread_log << " rows:" << number << std::endl;
      }
    mysql_free_result(result);
  }